### PR TITLE
FIX-#4100: Fall back to Pandas on row drop

### LINKED ISSUE #4100: ``Series.pop()`` doesn`t work correctly at MODIN_STORAGE_FORMAT=omnisci
When I run:
```
import modin.pandas as pd
from modin.pandas.test.utils import test_data_values
import modin.config as cfg
cfg.StorageFormat.put('omnisci')

data = test_data_values[0]
modin_series = pd.Series(data[next(iter(data.keys()))])
for key in modin_series.keys():
    modin_series.pop(key)
```
I have error:
```
AssertionError: Only column drop is supported
```

### PR DESCRIPTION
**Pull request by AndreyPavlenko**
Signed-off-by: Andrey Pavlenko <andrey.a.pavlenko@gmail.com>

## What do these changes do?
Only column drop is currently supported by the omnisci engine.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4100 
- [x] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/experimental/core/execution/native/implementations/omnisci_on_native/test/test_dataframe.py [hunk 1]
```diff
@@ -2140,6 +2140,18 @@ def drop_rename_concat(df1, df2, lib, **kwargs):
             force_arrow_execute=True,
         )
 
+    def test_drop_row(self):
+        def drop_row(df1, df2, lib, **kwargs):
```
Comment (ienkovich) [inline modin/experimental/core/execution/native/implementations/omnisci_on_native/test/test_dataframe.py:2144]: No need for unused args. You can use `def drop_row(df, **kwargs)` and skip `data2` for `run_and_compare`.

Comment (AndreyPavlenko) [reply modin/experimental/core/execution/native/implementations/omnisci_on_native/test/test_dataframe.py]: Fixed.

File: modin/experimental/core/execution/native/implementations/omnisci_on_native/test/test_dataframe.py [hunk 2]
```diff
@@ -2140,6 +2140,18 @@ def drop_rename_concat(df1, df2, lib, **kwargs):
...
         )
 
+    def test_drop_row(self):
+        def drop_row(df, **kwargs):
+            return df.drop(1)
+
+        run_and_compare(
+            drop_row,
+            data=self.data1,
+            data2=self.data2,
```
Comment (ienkovich) [inline modin/experimental/core/execution/native/implementations/omnisci_on_native/test/test_dataframe.py:2150]: Here you don't need to pass `data2` and `force_arrow_execute` which are unused by your test.

File: modin/experimental/core/storage_formats/hdk/query_compiler.py [hunk 3]
```diff
@@ -556,7 +556,8 @@ def concat(self, axis, other, **kwargs):
         return self.__constructor__(new_modin_frame)
 
     def drop(self, index=None, columns=None):
-        assert index is None, "Only column drop is supported"
+        if index is not None:
```
Comment (YarShev) [inline modin/experimental/core/storage_formats/hdk/query_compiler.py:559]: Why does the coverage not detect that we get into this branch? Do we have a test for this?

File: modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py [hunk 4]
```diff
@@ -2145,6 +2145,16 @@ def drop_rename_concat(df1, df2, lib, **kwargs):
             force_arrow_execute=True,
         )
 
+    def test_drop_row(self):
+        def drop_row(df, **kwargs):
+            return df.drop(1)
```
Comment (YarShev) [inline modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py:2150]: We should probably have a test that goes through the top level to the lower, i.e. I think we should have a test for `.pop` method, which goes through `.drop`? Also, `.pop` is used in the issue.

File: modin/experimental/core/storage_formats/hdk/query_compiler.py (hunk 3 above)
Comment (AndreyPavlenko) [reply modin/experimental/core/storage_formats/hdk/query_compiler.py:559]: I've checked in debugger and this code is executed by the test_drop_row() test. Removing this code, causes the test to fail.

File: modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py (hunk 4 above)
Comment (YarShev) [reply modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py:2150]: ping

File: modin/experimental/core/storage_formats/hdk/query_compiler.py (hunk 3 above)
Comment (YarShev) [reply modin/experimental/core/storage_formats/hdk/query_compiler.py:559]: ok, let's continue in this discussion ([link](https://github.com/modin-project/modin/pull/4937/files#r1029442629))

File: modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py (hunk 4 above)
Comment (YarShev) [reply modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py:2150]: Oh, I see that you added a test for `.pop`. Thanks

File: modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py [hunk 5]
```diff
@@ -2145,6 +2145,28 @@ def drop_rename_concat(df1, df2, lib, **kwargs):
...
```
Comment (YarShev) [inline modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py:2150]:
```suggestion
            return df.drop(labels=1)
```

**Review by YarShev — COMMENTED** [review]
Left a comment to be more explicit.

**Review by YarShev — APPROVED** [review]

**Pull request merged**